### ISSUE TITLE
check_icmp blasts packets out too fast

### ISSUE THREAD

**Issue by tognet** [nagios-plugins/nagios-plugins#636]
I love check_icmp over check_ping for being entirely self-contained, however, it only has one possible interval between packets no matter what you do: Pure flood ping. It does not care about any options (such as -i), it only has the one behavior for how fast it sends the pings out, which is to send the next ping as soon as a reply is received from the last one. Setting -n 20 to send 20 pings comes back with all 20 pings in 12ms when directed at a nearby host. That's quite the flood.

Measurements with check_icmp would work a lot better for more people if we could make the -i option function and honor the expected default of 80ms between packets.

I think the problem may be around line 917 in check_icmp.c.

**Comment by sawolf** [comment]
Yeah, it's interesting that there's a comment right in the section which explains that it's violating options set by the user. There should at least be a switch that disables the behavior.
Thanks for reporting this.

**Comment by sinusoidal-zz** [comment]
Just fell foul of this - use case was that I was attempting to duplicate check-ping ICMP behavior (5 pings, over 5 seconds), but with check-icmp, in order to make use of the potential performance benefits of native code.

I couldn't get -i  (max interval) or -I (max target interval) to have any impact (nor am I sure what the difference between them really is).
